src/services/services.ts

```diff
@@ -6099,6 +6197,25 @@ namespace ts {
                 const start = findInComments ? container.getFullStart() : container.getStart();
                 const possiblePositions = getPossibleSymbolReferencePositions(sourceFile, searchText, start, container.getEnd());
 
+                // If we are just looking for implementations and this is a property access expression, we need to get the
+                // symbol of the local type of the symbol the property is being accessed on. This is because our search
+                // symbol may have a different parent symbol if the local type's symbol does not declare the property
+                // being accessed (i.e. it is declared in some parent class or interface)
+                let parents: Symbol[];

```

Comment:
The compiler doesn't check this yet, but this is `Symbol[] | undefined` since it's only used if `implementations` is true, and so is its use in `getRelatedSymbol`, where it should be documented that it's only set for `implementations`.